internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1137,6 +1137,16 @@ func upgradeSettingsSchema() *pluginsdk.Schema {
 						Type:     pluginsdk.TypeString,
 						Required: true,
 					},
+					"drain_timeout_in_minutes": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+						Default:  30,
+					},
+					"node_soak_duration_in_minutes": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+						Default:  0,
```

Comment:
Integers that don't have a `Default` specified in the schema have a default of `0` anyway so this isn't necessary, what we can and should add here is the validation I mentioned earlier 
```suggestion
						ValidateFunc: validation.IntBetween(0, 30),
```
Apologies that wasn't clear earlier.